README.md

```diff
@@ -1,61 +1,274 @@
 
-	1. Search by id
 ````
-$response = $service->idsearch("",true, NULL, array("price","brand"));
+$service = new SearchService($access_key,$secret_key);
+$service->idsearch("blue_dress");
 ````
-	2. Search by color
+
+###5.2 Color Search
+Color search is to search images with similar color by providing a color code. The color code should be in Hexadecimal and passed to the colorsearch service.
+
 ````
-$response = $service->colorsearch("fa4d4d");
+$service->colorsearch("fa4d4d");
 ````
-	3. Search by uploading
-* image 
+
+###5.3 Upload Search
+Upload search is used to search similar images by uploading an image or providing an image url. `Image` class is used to perform the image encoding and resizing. You should construct the `Image` object and pass it to uploadsearch to start a search.
+
+Using an image from a local file path
+
 ````
-$image = new Image('test_file.png');
+$image = new Image(imagePath);
 $response = $service->uploadsearch($image);
 ````
-* url
+
+Alternatively, you can pass an image url directly to uploadsearch to start the search.
+
 ````
-$image = new Image('http://cdn-static.cnet.co.uk/i/product_media/40001253/image1/440x330-iphone-5-main.jpg');
+$image = new Image('http://mydomain.com/images/red_dress.jpg');
 $response = $service->uploadsearch($image);
-````
\ No newline at end of file
+````
+
+####5.3.1 Selection Box
+
+If the object you wish to search for takes up only a small portion of your image, or other irrelevant objects exists in the same image, chances are the search result could become inaccurate. Use the Box parameter to refine the search area of the image to improve accuracy. Noted that the box coordinated is setted with respect to the original size of the image passed, it will be automatically scaled to fit the resized image for uploading:
+
+```
+$image = new Image(imagePath);
+// create the box to refine the area on the searching image
+// Box(x1, y1, x2, y2) where (0,0) is the top-left corner
+// of the image, (x1, y1) is the top-left corner of the box,
+// and (x2, y2) is the bottom-right corner of the box.
+$box = new Box(0,0,10,10);
+$image->set_box($box);
+```
+####5.3.2 Resizing Settings
+When performing upload search, you may notice the increased search latency with increased image file size. This is due to the increased time spent in network transferring your images to the ViSearch server, and the increased time for processing larger image files in ViSearch. 
+
+To reduce upload search latency, by default the uploadSearch method makes a copy of your image file and resizes the copy to 512x512 pixels if both of the original dimensions exceed 512 pixels. This is the optimized size to lower search latency while not sacrificing search accuracy for general use cases:
+
+````
+$resizeSettings = new ResizeSettings();
+//default resize setting, set the image size to 512 x 512 with jpeg 85 quality
```

Comment:
update standard and high quality to 75, see https://github.com/visenze/visearch-sdk-java/blob/master/README.md#532-resizing-settings